mw-webapp/src/dataAccessLogic/DayReportDAL.ts

```diff
@@ -0,0 +1,191 @@
+import {dayReportToDayReportDTOConverter} from "src/dataAccessLogic/BusinessToDTOConverter/dayReportToDayReportDTOConverter";
+import {CurrentProblemDAL} from "src/dataAccessLogic/CurrentProblemDAL";
+import {dayReportDTOToDayReportConverter} from
+  "src/dataAccessLogic/DTOToBusinessConverter/dayReportDTOToDayReportConverter";
+import {JobDoneDAL} from "src/dataAccessLogic/JobDoneDAL";
+import {MentorCommentDAL} from "src/dataAccessLogic/MentorCommentDAL";
+import {PlanForNextPeriodDAL} from "src/dataAccessLogic/PlanForNextPeriodDAL";
+import {DayReport} from "src/model/businessModel/DayReport";
+import {DayReportService, NewDayReportProps} from "src/service/DayReportService";
+import {DateUtils} from "src/utils/DateUtils";
+
+/**
+ * Provides methods to interact with the DayReport business model
+ */
+export class DayReportDAL {
+
+  /**
+ * Day reports
+ * @returns {Promise<DayReport[]>}
+ */
+  public static async getDayReports(): Promise<DayReport[]> {
+    const dayReportsDTO = await DayReportService.getDayReportsDTO();
+    const jobsDonePreview = await JobDoneDAL.getJobsDone();
+    const plansForNextPeriodPreview = await PlanForNextPeriodDAL.getPlansForNextPeriod();
+    const mentorCommentsPreview = await MentorCommentDAL.getMentorComments();
+    const problemsForCurrentPeriodPreview = await CurrentProblemDAL.getCurrentProblems();
+    const firstReport = dayReportsDTO[0];
+
+    const jobsDone = firstReport.jobsDone.map((jobDoneUuid) => {
+      const jobDone = jobsDonePreview
+        .find((elem) => elem.uuid === jobDoneUuid);
```

Comment:
Find inside map is too slow, refactor using hash map

Reply:
Created task #114

Reply:
Why?

Reply:
Refactoring required. Don't see  reasons to make it later